src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java

```diff
@@ -358,28 +369,71 @@ public void validate() {
+        return hasValidationAndLoadErrors() ||
+            hasFeedVersionExpired() ||
+            hasHighSeverityErrorTypes();
     }
 
     /**
-     * Does this feed have any critical errors other than possibly being expired?
-     * @return whether feed version has critical errors (outside of expiration)
+     * Does this feed have any validation or load errors?
+     * @return whether feed version has any validation or load errors
      */
-    private boolean hasCriticalErrorsExceptingDate() {
+    private boolean hasValidationAndLoadErrors() {
         if (validationResult == null)
             return true;
-
         return validationResult.fatalException != null ||
             !validationSummary().bounds.areValid() ||
             feedLoadResult.stopTimes.rowCount == 0 ||
             feedLoadResult.trips.rowCount == 0 ||
             feedLoadResult.agency.rowCount == 0;
+    }
+
+    /**
+     * Has this feed expired?
+     * @return If the validation result last calendar date is null or has expired return true, else return false.
+     */
+    private boolean hasFeedVersionExpired() {
+        return validationResult.lastCalendarDate == null ||
+            LocalDate.now().isAfter(validationResult.lastCalendarDate);
+    }
 
+    /**
+     * Has this feed version produced any high severity error types when being validated? Some very rural services have
+     * high priority errors for valid stops that aren't near population centers. These acceptable high priority errors
+     * are ignored.
+     * @return whether high severity error types have been flagged.
+     */
+    private boolean hasHighSeverityErrorTypes() {
+        Set<String> acceptableHighSeverityErrorTypes
+            = new HashSet<>(Arrays.asList(
+                NewGTFSErrorType.ROUTE_UNUSED.name(),
+                NewGTFSErrorType.STOP_GEOGRAPHIC_OUTLIER.name(),
+                NewGTFSErrorType.STOP_LOW_POPULATION_DENSITY.name()));
+        Set<String> highSeverityErrorTypes = Stream.of(NewGTFSErrorType.values())
+                .filter(type -> type.priority == Priority.HIGH &&
+                    !acceptableHighSeverityErrorTypes.contains(type.name()))
+                .map(NewGTFSErrorType::toString)
+                .collect(Collectors.toSet());
+        try (Connection connection = GTFS_DATA_SOURCE.getConnection()) {
+            String sql = String.format("select distinct error_type from %s.errors", namespace);
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            LOG.info(preparedStatement.toString());
+            ResultSet resultSet = preparedStatement.executeQuery();
+            // Check if any of the error types found in the table are "high priority/severity"
+            while (resultSet.next()) {
+                String errorType = resultSet.getString(1);
+                if (highSeverityErrorTypes.contains(errorType)) return true;
```

Comment:
Convert to a `where` clause in the SQL statement, and you can even just limit the results to one row.

Reply:
I had this initially to return a count of high severity error types using a where clause, but it was deemed an sql injection risk and the where clause was becoming a bit combersome to construct using a prepared statement. Short answer, Landon sugested this approach instead!